pkg/ingester/errors_test.go

```diff
@@ -239,23 +242,44 @@ func TestErrorWithStatus(t *testing.T) {
 	err := newSampleTimestampTooOldError(timestamp, metricLabelAdapters)
 	errWithStatus := newErrorWithStatus(err, codes.Unavailable)
 	require.Error(t, errWithStatus)
+	// Ensure gogo's status.FromError recognizes errWithStatus.
 	stat, ok := status.FromError(errWithStatus)
 	require.True(t, ok)
 	require.Equal(t, codes.Unavailable, stat.Code())
 	require.Errorf(t, err, stat.Message())
 	require.Empty(t, stat.Details())
+
+	// Ensure grpc's status.FromError recognizes errWithStatus.
+	st, ok := grpcstatus.FromError(errWithStatus)
+	require.True(t, ok)
+	require.Equal(t, codes.Unavailable, st.Code())
+	require.Errorf(t, err, st.Message())
+
+	// Ensure httpgrpc's HTTPResponseFromError does not recognize errWithStatuserrWithStatus.
+	resp, ok := httpgrpc.HTTPResponseFromError(errWithStatus)
```

Comment:
why do we want it to not recognize the status? is that just to ensure that no behavior changes?

Reply:
`httpgrpc.HttpResponseFromError()` ensures that the given error is built by `httpgrpc.Errorf()`. Since we need to guarantee backwards compatibility, methods that translate errors to the corresponding HTTP or gRPC codes will need to be able to distinguish between errors built by `httpgrpc.Errorf()` and `status.Error()`. 
`status.FromError()` recognizes both types of errors, but `httpgrpc.HttpResponseFromError()` recognzies only the former. This test ensures that this property holds.